pubsub/azure/eventhubs/eventhubs.go

```diff
@@ -98,62 +117,301 @@ func subscribeHandler(ctx context.Context, topic string, e *eventhub.Event, hand
 
 // AzureEventHubs allows sending/receiving Azure Event Hubs events.
 type AzureEventHubs struct {
-	hub      *eventhub.Hub
-	metadata azureEventHubsMetadata
-
-	logger        logger.Logger
-	ctx           context.Context
-	cancel        context.CancelFunc
-	backOffConfig retry.Config
+	metadata           *azureEventHubsMetadata
+	logger             logger.Logger
+	ctx                context.Context
+	cancel             context.CancelFunc
+	backOffConfig      retry.Config
+	hubClients         map[string]*eventhub.Hub
+	eventProcessors    map[string]*eph.EventProcessorHost
+	hubManager         *eventhub.HubManager
+	eventHubSettings   azauth.EnvironmentSettings
+	managementSettings azauth.EnvironmentSettings
+	cgClient           *mgmt.ConsumerGroupsClient
+	tokenProvider      *aad.TokenProvider
 }
 
 type azureEventHubsMetadata struct {
-	connectionString     string
-	consumerGroup        string
-	storageAccountName   string
-	storageAccountKey    string
-	storageContainerName string
+	ConnectionString        string `json:"connectionString,omitempty"`
+	EventHubNamespace       string `json:"eventHubNamespace,omitempty"`
+	ConsumerGroup           string `json:"consumerID"`
```

Comment:
Why JSON is consumerID and not ConsumerGroup? I think these should match, not?

Reply:
I think consumerID is a field coming from dapr ... It is across all pubsub ... not only eventhub ... And in every pubsub it is mapped to some form of a field called cosumer group or such ...

Reply:
The metadata should make sense within the component itself. Someone that uses EventHubs expects ConsumerGroup as the terminology. There is no requirement that metadata should match cross components.

Reply:
Currently the metadata properties are marshalled into json and then un-marshalled into the Go struct here. `consumerID` is the incoming property name from `daprd` Here `consumerID` is automatically set by `daprd` and not an independent property of this component. 
It is called consumerGroup within this component.

Reply:
OK. I found the code that handles this logic in daprd. That is the logic that automatically sets the ConsumerID attributed based on the app ID. So, this attribute is the same for all components. LGTM.